### ISSUE TITLE
Change all metrics methods from blocking to async

### ISSUE THREAD

**Issue by chia7712** [oharastream/ohara#678]
@saivirtue noticed that the beans APIs are all in blocking mode and it is not acceptable to current arch of ohara (see ohara-metrics).

**Comment by saivirtue** [comment]
Do we need async? Since metrics data is time-related, I think we could "get the previous value" if caller cannot get the response in time

**Comment by chia7712** [comment]
> Do we need async? Since metrics data is time-related, I think we could "get the previous value" if caller cannot get the response in time

Do you mean "local cache"? we still need to connect to "remote" node to get metrics and the "connection" is a blocking call now.

**Comment by saivirtue** [comment]
> Do you mean "local cache"?

Yep. async call will make the time-series data a little weird (ex : t1 data response after t2 had been received, and the chart will look like)

            v2       v1
-------------------->
t1         t2       t1'

**Comment by chia7712** [comment]
> Yep. async call will make the time-series data a little weird (ex : t1 data response after t2 had been received)

Personally, caller (or user) SHOULD NOT care for the consistency (or happen-before) of metrics. It is ok to have a simple async implementation.
